source/Windows.Devices.I2c/I2cDevice.cs

```diff
@@ -27,11 +33,26 @@ internal I2cDevice(string i2cBus, I2cConnectionSettings settings)
             // the encoding is (I2C bus number x 1000 + slave address)
             // i2cBus is an ASCII string with the bus name in format 'I2Cn'
             // need to grab 'n' from the string and convert that to the integer value from the ASCII code (do this by subtracting 48 from the char value)
-            var deviceId = (i2cBus[3] - 48) * 1000 + settings.SlaveAddress;
+            var controllerId = i2cBus[3] - 48;
+            var deviceId = (controllerId * 1000) + settings.SlaveAddress;
+
+            I2cController controller;
+
+            if (!I2cControllerManager.ControllersCollection.Contains(controllerId))
+            {
+                // this controller doesn't exist yet, create it...
+                controller = new I2cController(i2cBus);
```

Comment:
Don't we need to add the created controller to the I2cControllerManager.ControllersCollection (thread-safe)?

Reply:
This is already done inside the I2cController constructor.